erpnext/accounts/report/balance_sheet/balance_sheet.py

```diff
@@ -49,17 +66,62 @@ def execute(filters=None):
 		unclosed["total"]=opening_balance
 		data.append(unclosed)
 
-	if provisional_profit_loss:
-		data.append(provisional_profit_loss)
 	if total_credit:
+		data.append({})
 		data.append(total_credit)
 
 	columns = get_columns(filters.periodicity, period_list, filters.accumulated_values, company=filters.company)
-
 	chart = get_chart_data(filters, columns, asset, liability, equity)
 
 	return columns, data, message, chart
 
+def insert_equity_provisions(equity, provisional_profit_loss, period_list):
+	"""
+	[summary]
+
+	Args:
+		equity (list): dicts containing equity accounts and their periodic balances
+		provisional_profit_loss (dict): details of provisional profit / loss
+		period_list (list): list of periods for which the data needs to be processed
+
+	Returns:
+		dict: details of equity account containing provisional profits / loss
+	"""
+	total_equity = equity[-2]
+	provisional_profit_loss["indent"] = total_equity.get("indent") + 1
+	equity.insert(-2, provisional_profit_loss) #add provisions to equity list
+	equity.pop()
+	for period in period_list:
+		if period.key in total_equity and period.key in provisional_profit_loss:
+			total_equity[period.key] = 0.0 if not total_equity[period.key] else total_equity[period.key]
+			total_equity[period.key] += provisional_profit_loss[period.key] #update equity account total for all periods
+	return equity
+
+
+
+def create_equity_provisions(provisional_profit_loss, period_list):
```

Comment:
`insert_equity_provisions` and `create_equity_provisions` sounds almost similar, can you use some better naming for these functions?

Reply:
done.